src/software/thunderscope/thunderscope_main.py

```diff
@@ -331,6 +332,13 @@ def __async_sim_ticker(tick_rate_ms):
             :param tick_rate_ms: The tick rate of the simulation
 
             """
+            sim_start_buffer = ThreadSafeBuffer(1, SimulationStartedTrigger, False)
+            tscope.simulator_proto_unix_io.register_observer(
+                SimulationStartedTrigger, sim_start_buffer
+            )
+
+            sim_start_buffer.get(block=False)
```

Comment:
We want this function to be blocked (i.e. not do anything) until this buffer has received the `SimulationStartedTrigger` message from C++. Then we will send the `WorldState` to C++ since we know for a fact that it's setup.
```suggestion
            sim_start_buffer.get(block=True)
```
Adding `block=True` does exactly that. It stops execution of this function until the buffer has a value in it for it to `get`.